src/redux/crypto/cryptoSlice.js

```diff
@@ -0,0 +1,72 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const fetchCrypto = createAsyncThunk(
+  'crypto/fetchCrypto',
+  async () => {
+    const response = await fetch('https://api.coinstats.app/public/v1/coins/');
+    const data = await response.json();
+    const coins = data.coins.map((coin) => ({
+      id: coin.id,
+      icon: coin.icon,
+      name: coin.name,
+      symbol: coin.symbol,
+      rank: coin.rank,
+      price: coin.price,
+      volume: coin.volume,
+      marketCap: coin.marketCap,
+      availableSupply: coin.availableSupply,
+      totalSupply: coin.totalSupply,
+      pricechange1h: coin.priceChange1h,
+      pricechange1d: coin.priceChange1d,
+      pricechange1w: coin.priceChange1w,
+      url: coin.websiteUrl,
+      show: false,
+
+    }));
+    return coins;
+  },
+);
+
+const cryptoSlice = createSlice({
+  name: 'crypto',
+  initialState: {
+    isLoding: false,
+    crypto: [],
+    isEror: false,
+  },
+  reducers: {
+    getdetails: (state, action) => {
+      const id = action.payload;
+      const coindata = state;
+      coindata.crypto = state.crypto.map((coin) => {
+        const coins = coin;
+        if (coin.id === id) {
+          coins.show = true;
+        } else {
+          coins.show = false;
+        }
+        return coin;
+      });
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(fetchCrypto.pending, (state) => {
+      const Loding = state;
+      Loding.isLoding = true;
+    });
```

Comment:
- [ ] Good job so far 👏 You have really done a great job up to now. I would also say that as suggested by the previous reviewers, you really need to test the pure function found here in the reducers. I think maybe It is because they have not provided you with all the important information. As you know, pure functions are functions that when given the same input will always give the same output. They don't have any external dependency, just the arguments passed in. 
From the project requirements, basically, **they want you to write tests for your actions as found here in the reducer object of the cryptoSlice** One action function you can test for example is `getDetails` and if you are wondering how to test reducer actions, the following resources may be helpful:
- The tutorial documentation [here](https://medium.com/asos-techblog/how-to-test-your-react-redux-application-48d90481a253)
- The Test Example [here](https://github.com/BertrandConxy/Stocks-market-trends/blob/dev-branch/src/__tests__/PureFunctions.test.js)

I hope all of those will be helpful 😃 😉